### PR TITLE
Network Plugin - Update UI for routes 

### PR DESCRIPTION
**Pull request by bizzguy**
## Summary

The current UI for managing mock network request definitions (routes) has some issues
- too small for the amount of information it displays
- no formatting for JSON response bodies
- uses the older design system (not ant)
- no separation between commands for creating routes and the list of routes

The following screen images show these problems.

image

image

## Changelog

Enhance UI for Mocks dialog in Network Plugin

## Test Plan

Ran multiple manual tests using the Android sample app.  Here is a screen image (at 1280 x 720) for the enhanced UI.

image

### CODE REVIEW

**Comment by bizzguy** [comment]
@mweststrate  I'm also working on the export/import feature for mocks and wanted to make these UI changes as a separate PR.

I hard-coded the dialog height/width but made sure it still fit on a 1280 x 720 screen.  I also added "Layout.Container" which to some of the components (this is more "ant" compliant, right?).

Any suggestions would be appreciated.

**Comment by bizzguy** [comment]
This PR also seems to fix the problem with adding routes described here https://github.com/facebook/flipper/issues/1733

File: desktop/plugins/network/MockResponseDetails.tsx [hunk 1]
```diff
@@ -322,31 +349,11 @@ export function MockResponseDetails({id, route, isDuplicated}: Props) {
               stickyBottom={true}
               autoHeight={true}
               floating={false}
-              // height={300}
```
Comment (mweststrate) [inline desktop/plugins/network/MockResponseDetails.tsx:325]: Does this fix https://github.com/facebook/flipper/issues/1460? Otherwise Wrapping a `Layout.ScrollContainer` could fix that I'd expect

File: desktop/plugins/network/ManageMockResponsePanel.tsx [hunk 2]
```diff
@@ -248,11 +259,13 @@ export function ManageMockResponsePanel(props: Props) {
       </LeftPanel>
       <RightPanel>
         {selectedId && props.routes.hasOwnProperty(selectedId) && (
+          <Layout.ScrollContainer vertical style={{height: 500}}>
```
Comment (mweststrate) [inline desktop/plugins/network/ManageMockResponsePanel.tsx:262]: Why did you need a ScrollContainer here? To fix the header pane? In that case I'd move the scrolling inside the tab, so that the tab switcher stays always visible.

File: desktop/plugins/network/ManageMockResponsePanel.tsx [hunk 3]
```diff
@@ -195,7 +199,7 @@ export function ManageMockResponsePanel(props: Props) {
     props.routes,
   ]);
   return (
-    <Container>
+    <Container style={{height: 580}}>
```
Comment (mweststrate) [inline desktop/plugins/network/ManageMockResponsePanel.tsx:202]: Note that we now expose `<Layout.Left`, so I suspect the locally styled components aren't largely not needed anymore if you do something like:

```
<Layout.Container height={580}> 
    <Layout.Left>
       <Layout.Container width={200}>
          { /* Left content */ }
        </Layout.Container>
        { /*Right content, probably inside container as well */ }
      </Layout.Left>
 </Layout.Container>
 ```


But if you want to keep the changes small, this is fine as well. Note that you can play with the new layout abstractions by using `View > Flipper Design System` to see some examples.

File: desktop/plugins/network/ManageMockResponsePanel.tsx [hunk 4]
```diff
@@ -217,14 +221,21 @@ export function ManageMockResponsePanel(props: Props) {
               props.responses,
             );
           }}>
-          <Glyph
-            name="plus-circle"
-            size={16}
-            variant="outline"
-            color={colors.blackAlpha30}
-          />
+            <Glyph
```
Comment (mweststrate) [inline desktop/plugins/network/ManageMockResponsePanel.tsx:224]: Looks like prettier wasn't run on this PR. VSCode (and most other IDEs have a plugin for it). If you have trouble running it, let us know, we can run it on our side as well.

**Review by mweststrate — COMMENTED** [review]
Hey @bizzguy, thanks for looking into this! Left a few comments.

**Comment by passy** [comment]
Sorry, don't have anything useful to say about the code just wanted to swoop in to say that this is awesome! Thanks for doing this. :)

**Comment by priteshrnandgaonkar** [comment]
@bizzguy, does this PR also tackle this [issue](https://github.com/facebook/flipper/issues/1487).

File: desktop/plugins/network/MockResponseDetails.tsx (hunk 1 above)
Comment (bizzguy) [reply desktop/plugins/network/MockResponseDetails.tsx:325]: Yes, this does resolve https://github.com/facebook/flipper/issues/1460.

File: desktop/plugins/network/ManageMockResponsePanel.tsx (hunk 2 above)
Comment (bizzguy) [reply desktop/plugins/network/ManageMockResponsePanel.tsx:262]: This ScrollContainer can be removed.  Done.

File: desktop/plugins/network/ManageMockResponsePanel.tsx (hunk 4 above)
Comment (bizzguy) [reply desktop/plugins/network/ManageMockResponsePanel.tsx:224]: Is "prettier" the same as "yarn fix" (which I did forget to run)?

**Comment by bizzguy** [comment]
@priteshrnandgaonkar   This PR does not solve the problem of saving the routes properly.  But I am working on that along with a few other Route features.

File: desktop/plugins/network/ManageMockResponsePanel.tsx (hunk 3 above)
Comment (bizzguy) [reply desktop/plugins/network/ManageMockResponsePanel.tsx:202]: @mweststrate   I would like to keep this change small for now.  Once this is approved I'm planning on putting in the PR for import/export of routes.  Then I'd like to revisit the route UI and add a checkbox to routes for enable/disable.
